src/components/qrModal/qrModalView.tsx

```diff
@@ -97,42 +114,145 @@ export const QRModal = ({}: QRModalProps) => {
 
   const onSuccess = (e) => {
     setIsScannerActive(false);
-    _handleDeepLink(e.data);
+    if (isHiveUri(e.data)) {
+      _handleHiveUri(e.data);
+    } else {
+      _handleDeepLink(e.data);
+    }
   };
 
-  const _handleDeepLink = async (url) => {
-    setIsProcessing(true);
-    const deepLinkData = await deepLinkParser(url, currentAccount);
-    const { name, params, key } = deepLinkData || {};
-    setIsProcessing(false);
-    if (name && params && key) {
+  const _handleHiveUri = async (uri: string) => {
+    try {
       setIsScannerActive(false);
       _onClose();
-      RootNavigation.navigate(deepLinkData);
-    } else {
-      Alert.alert(
-        intl.formatMessage({ id: 'qr.unsupported_alert_title' }),
-        intl.formatMessage({ id: 'qr.unsupported_alert_desc' }),
-        [
+      if (!isLoggedIn) {
+        showLoginAlert({ intl });
+        return;
+      }
+      if (isPinCodeOpen) {
+        RootNavigation.navigate({
+          name: ROUTES.SCREENS.PINCODE,
+          params: {
+            callback: () => _handleHiveUriTransaction(uri),
+          },
+        });
+      } else {
+        _handleHiveUriTransaction(uri);
+      }
+    } catch (err) {
+      _showInvalidAlert();
+    }
+  };
+
+  const _handleHiveUriTransaction = async (uri: string) => {
+    if (get(currentAccount, 'local.authType') === authType.STEEM_CONNECT) {
+      await delay(500); // NOTE: it's required to avoid modal mis fire
+      dispatch(
+        showWebViewModal({
+          uri: uri,
+        }),
+      );
+      return;
+    }
+
+    const parsed = hiveuri.decode(uri);
+    // resolve the decoded tx and params to a signable tx
+    let { tx, signer } = hiveuri.resolveTransaction(parsed.tx, parsed.params, {
+      signers: currentAccount.name,
+      preferred_signer: currentAccount.name,
+    });
+    const operations = get(tx, 'operations', []);
+    dispatch(
+      showActionModal({
+        title: intl.formatMessage({
+          id: 'qr.confirmTransaction',
+        }),
+        bodyContent: _checkOpsArray(operations) ? _renderActionModalBody(operations[0]) : null,
```

Comment:
@aliseyalvi according to this, if checkOpsArray fails, operation will still continue upon confirmation, only this time user will be confirming blindly, wouldn't it make is counter intuitive.. why exactly cannot we show multiple operations array....? or perhaps if multiple ops detected, we can let user know app do no currently support singing multiple operations in a single transaction yet.

Reply:
@noumantahir change request is done here https://github.com/ecency/ecency-mobile/pull/2713/commits/55069cd74a95726f9120972531399d6958f7f334